From ab9a1bb945b3262bd2b0fb90c3678c251d2c72cf Mon Sep 17 00:00:00 2001
From: Johannes Schindelin <johannes.schindelin@gmx.de>
Date: Mon, 2 Mar 2015 14:09:29 +0300
Subject: [PATCH 12/56] convert(): plug potential memory leak

When the result of conversion is identical to the original argument, we
should make sure that the temporary buffer was released.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/path.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/winsup/cygwin/path.cc b/winsup/cygwin/path.cc
index d15127e..f265131 100644
--- a/winsup/cygwin/path.cc
+++ b/winsup/cygwin/path.cc
@@ -3464,6 +3464,10 @@ arg_heuristic_with_exclusions (char const * const arg, char const * exclusions,
   // Don't allocate memory if no conversion happened.
   if (!strcmp (arg, stack_path))
     {
+      if (arg != stack_path)
+        {
+          free (stack_path);
+        }
       return ((char *)arg);
     }
   arg_result = (char *)realloc (stack_path, strlen (stack_path)+1);
-- 
2.9.0

